docs/proposals/dynamic-kubelet-configuration.md

```diff
@@ -0,0 +1,171 @@
+    + Some additional thought needs to go toward the graduation policy for experimental fields. Will the kubelet continue to try to parse them out of the experimental section? For how long? What if the first-class field and experimental field conflict?
+
+
+### Representation and Organization of Kubelet Configuration in a Cluster
+
+- As a cluster-level object, Kubelet's configuration should be stored in a `ConfigMap` object.
+- On local disk, the Kubelet's configuration should be stored in a `.json` or `.yaml` file.
+- The Kubelet's configuration should be, at least initially, organized in the cluster as a structured monolith. 
+    + *Structured*, so that it is readable.
+    + *Monolithic*, to provide atomicity over the entire configuration object.
+        * This likely means that the Kubelet configuration will be stored as a string blob (JSON or YAML) in the value associated with a given key on a `ConfigMap`.
+        * If leaky boundaries occur between the substructures, we don't want the problem of coordinating non-atomic updates across separately-referenced substructures.
+        * If, in the future, the ability to independently orchestrate the substructures of the configuration is desired, we can move down that road. But today this is probably overkill, because most K8s cluster configuration is eventually homogeneous anyway. Even in a non-homogeneously configured cluster, we would have to carefully consider the downside of losing atomicity on the config object against the upside of more flexibility for splitting up configuration responsibility and e.g. being able to roll out changes to separate subcomponents at different rates.
+
+An example of what the `ConfigMap` object for just the Kubelet could look like:
+```
+kind: ConfigMap
+metadata:
+  name: my-kubelet-config-<hash of `data` for verification>
+data:
+ config: "{JSON representation of config, for example}"
+```
+
+If this idea extended so that all `Node` config is packaged together:
+```
+kind: ConfigMap
+metadata:
+  name: my-node-config-<hash of `data` for verification>
+data:
+ kubelet: "{JSON representation of config, for example}"
+ kube-proxy: "{JSON representation of config, for example}"
+```
+
+### Referencing Configuration
+
+How do you tell the kubelet what to use?
+
+- On disk: The configuration should be specified via a path passed to a command-line flag. This should be an absolute path.
+- Cluster level object: The `ConfigMap` containing the desired configuration should be specified via the `Node` object corresponding to the Kubelet. Eventually, the `Node` may even have a `NodeConfig` field, but we ought to start by using an annotation, because additions to GA APIs are hard to change later if we get them wrong.
```

Comment:
What is the bootstrap path here? e.g., it starts with what it finds on disk, joins the cluster, downloads the current config and writes that to disk in case it restarts? Or something else?